csrc/host_ir/executor.cpp

```diff
@@ -329,20 +329,35 @@ void HostIrEvaluator::handle(LaunchKernel* launch_kernel) {
   for (auto& input : launch_kernel->inputs()) {
     args.push(getKnownConcreteValue(input));
   }
+
+  // If all output buffers are known already, pass them to the executor
```

Comment:
IIUC from your other comment, outputs will be either all allocated by KernelExecutor (as is) or all preallocated by HostIrEvaluator (in short future), and we don't expect a mixture of preallocated and non-preallocated outputs. 

```
for each output {
  if is known {
    outputs.push_back(the known output tensor)
  }
}
NVF_ERROR(outputs.empty() || outputs.size() == launch_kernel->outputs().size());
```
This way, a mixture of preallocated and allocated will be reported as a bug.